ports/mimxrt/boards/ADAFRUIT_METRO_M7/mpconfigboard.h

```diff
@@ -0,0 +1,65 @@
+#define MICROPY_HW_BOARD_NAME "Adafruit Metro M7"
+#define MICROPY_HW_MCU_NAME   "MIMXRT1011DAE5A"
+
+// i.MX RT1010 EVK has 1 board LED
+#define MICROPY_HW_LED1_PIN (pin_GPIO_03)
+#define MICROPY_HW_LED_ON(pin) (mp_hal_pin_high(pin))
+#define MICROPY_HW_LED_OFF(pin) (mp_hal_pin_low(pin))
+
+#define MICROPY_HW_NUM_PIN_IRQS (2 * 32)
+
+// Define mapping logical UART # to hardware UART #
+// LPUART1 on USB_DBG  -> 0
+// LPUART1 on D0/D1    -> 1
+// LPUART3 on A0/D4    -> 3
+// LPUART4 on D6/D7    -> 2
+
+#define MICROPY_HW_UART_NUM     (sizeof(uart_index_table) / sizeof(uart_index_table)[0])
+#define MICROPY_HW_UART_INDEX   { 1, 1, 2, 4 }
+
+#define IOMUX_TABLE_UART \
+    { IOMUXC_GPIO_10_LPUART1_TXD }, { IOMUXC_GPIO_09_LPUART1_RXD }, \
+    { IOMUXC_GPIO_12_LPUART3_TXD }, { IOMUXC_GPIO_11_LPUART3_RXD }, \
+    { 0 }, { 0 }, \
+    { IOMUXC_GPIO_AD_02_LPUART4_TXD }, { IOMUXC_GPIO_AD_01_LPUART4_RXD },
+
+#define MICROPY_HW_SPI_INDEX { 1 }
+
+#define IOMUX_TABLE_SPI \
+    { IOMUXC_GPIO_AD_06_LPSPI1_SCK }, { IOMUXC_GPIO_AD_05_LPSPI1_PCS0 }, \
+    { IOMUXC_GPIO_AD_04_LPSPI1_SDO }, { IOMUXC_GPIO_AD_03_LPSPI1_SDI }, \
+    { IOMUXC_GPIO_AD_02_LPSPI1_PCS1 }
+
+#define DMA_REQ_SRC_RX { 0, kDmaRequestMuxLPSPI1Rx, kDmaRequestMuxLPSPI2Rx }
+#define DMA_REQ_SRC_TX { 0, kDmaRequestMuxLPSPI1Tx, kDmaRequestMuxLPSPI2Tx }
+
+// Define mapping hardware I2C # to logical I2C #
+// SDA/SCL  HW-I2C    Logical I2C
+// D14/D15  LPI2C1 ->    0
+// D0/D1    LPI2C2 ->    1
+// D6/D7    LPI2C2 ->    1   Alternatively possible GPIO_AD_01, GPIO_AD_02
+
+#define MICROPY_HW_I2C_INDEX   { 1, 2 }
+
+#define IOMUX_TABLE_I2C \
+    { IOMUXC_GPIO_02_LPI2C1_SCL }, { IOMUXC_GPIO_01_LPI2C1_SDA }, \
+    { IOMUXC_GPIO_10_LPI2C2_SCL }, { IOMUXC_GPIO_09_LPI2C2_SDA },
+
+// Wifi Deinitions
+#define MICROPY_PY_NETWORK_HOSTNAME_DEFAULT "mpy-metro-m7"
+
+#define MICROPY_HW_WIFI_SPI_ID       (0)
+#define MICROPY_HW_WIFI_SPI_BAUDRATE (8000000)
+
+#define MICROPY_HW_NINA_ACK          pin_find(mp_obj_new_str("ESP_BUSY", 8))
+#define MICROPY_HW_NINA_CS           pin_find(mp_obj_new_str("ESP_CS", 6))
+#define MICROPY_HW_NINA_RESET        pin_find(mp_obj_new_str("ESP_RESET", 9))
```

Comment:
I'm not sure this is the right way to get a pin, it's quite inefficient.  Why not just use `&pin_GPIO_AD_07`?

Reply:
I've had chosen that way for better readability in the file. This part of the code is only executed once while the interface is set-up, so run-time efficiency does not matter. It requires a little bit more code, which is not a problem at the M7 board. But I can of course change it.

Reply:
Would this work: `pin_find(MP_QSTR_ESP_BUSY)`?  At least then it doesn't need to allocate the string.

Or maybe mimxrt needs named pins like stm32, then you could do `pyb_pin_ESP_BUSY` (probably a better name though, not pyb).

Reply:
Actually it should be `pin_find(MP_OBJ_NEW_QSTR(MP_QSTR_ESP_BUSY))`.

Reply:
The latter works, and WiFi starts with it.

> Or maybe mimxrt needs named pins like stm32, then you could do pyb_pin_ESP_BUSY (probably a better name though, not pyb).

At the moment the pin names as like `pin_GPIOxxxxx`. But reworking the Pin section is another huge task on the t.b.d. list.

Reply:
OK, that looks good then.  Happy to merge this now.

> But reworking the Pin section is another huge task on the t.b.d. list.

I'd suggest holding back on that.  I'd rather consolidate all ports to use the same pins.csv mechanism.

Reply:
> I'd suggest holding back on that. I'd rather consolidate all ports to use the same pins.csv mechanism.

That looks huge.

Reply:
I'm not suggesting that you do it.  It was on @jimmo 's list for a while now.